src/bootstrap/download.rs

```diff
@@ -474,7 +474,25 @@ impl Config {
             None
         };
 
-        self.download_file(&format!("{base_url}/{url}"), &tarball, "");
+        let mut help_on_error = "";
+        let buf;
+        if destination == "ci-rustc" {
+            help_on_error = "error: failed to download pre-built rustc from CI
+
+note: old builds get deleted after a certain time
+help: if trying to compile an old commit of rustc, disable `download-rustc` in config.toml:
+
+[rust]
+download-rustc = false
+";
+            if t!(std::fs::read(self.src.join("src/ci/channel"))) != b"nightly\n" {
+                buf = format!(
+                    "{help_on_error}\nhelp: if you are trying to contribute to rust-lang/rust, build from the latest `master` branch, not a published tag"
```

Comment:
You should be able to build on beta and stable branches, the retention on artifacts is long enough IIRC.

So I think I would just drop this help, it's probably not very helpful in practice.

Reply:
ok, done